src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -179,7 +184,16 @@ private async Task ExecuteRequests(Hl7.Fhir.Model.Bundle responseBundle, Hl7.Fhi
                     entryComponent.Response = new Hl7.Fhir.Model.Bundle.ResponseComponent { Status = ((int)HttpStatusCode.NotFound).ToString() };
                 }
 
-                responseBundle.Entry.Add(entryComponent);
+                if (responseBundle.Entry?.Any() != true)
```

Comment:
Would it be sufficient to only do `if(responseBundle.Entry != null)`?

Reply:
I wrote this first but then found out.. responseBundle.Entry is not null but count is zero.. so the statement validates for both (null and count). And if Entry list is null or count is zero.. I created a list with the capacity of requests size

Reply:
This does not need to happen in the foreach loop right? It could just happen once before the loop?